neuros/board/stream_processor.py

```diff
@@ -0,0 +1,136 @@
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Iterator, Optional, Callable, Any
+import numpy as np
+import threading
+import time
+from brainflow.board_shim import BoardShim, BrainFlowInputParams, BoardIds
+
+# Global lock for thread-safe board operations
+_board_lock = threading.Lock()
+
+
+@contextmanager
+def board_stream(board_id: int = BoardIds.SYNTHETIC_BOARD) -> Iterator[BoardShim]:
```

Comment:
Pretty sure that it's not returning an Iterator, but even if it is, that's the wrong data type. We don't have a stream of boards, we have one board with a stream of data

Reply:
You are correct. It should return a BoardShim. Fixing.